src/components/PageHeader.astro

```diff
@@ -0,0 +1,82 @@
+<header>
+  <div class="container flex fixed justify-center !max-w-full !p-0 z-10">
+    <nav class="flex flex-col h-10 w-screen bg-sky-500">
+      <div
+        class="flex min-w-full min-h-full justify-end items-center md:hidden"
+      >
+        <button
+          id="mobile-navigation"
+          class="mr-2 cursor-pointer flex items-center justify-center"
+        >
+          <div class="nav-bars">
+            <div class="nav-bar"></div>
+            <div class="nav-bar"></div>
+            <div class="nav-bar"></div>
+          </div>
+        </button>
+      </div>
+
+      <div class="flex justify-center">
+        <ul
+          id="navigation-list"
+          class="nav-close flex flex-col justify-around items-center w-screen h-screen text-3xl bg-slate-100 duration-500 md:flex-row md:h-10 md:max-w-6xl md:justify-end md:bg-inherit md:text-base"
+        >
+          <li class="m-10"><a href="#home">Home</a></li>
+          <li class="m-10"><a href="#portfolio">Portfolio</a></li>
+          <li class="m-10"><a href="#tbd">TBD</a></li>
+        </ul>
+      </div>
+    </nav>
+  </div>
+</header>
+
+<style>
+  .nav-bar {
+    @apply bg-black h-[2px] transform transition-all origin-left;
+  }
+
+  .nav-bars {
+    @apply flex flex-col justify-between w-[26px] h-5;
+  }
+
+  .nav-close {
+    @apply translate-x-full md:translate-x-0;
+  }
+
+  body[data-navigation-open="true"] {
+    @apply overflow-y-hidden;
+
+    .nav-bar:nth-child(1) {
+      @apply rotate-[45deg];
+    }
+    .nav-bar:nth-child(2) {
+      @apply hidden;
+    }
+    .nav-bar:nth-child(3) {
+      @apply -rotate-[45deg];
+    }
+  }
+</style>
+
+<script>
+  function burgerList() {
+    const mobileIcon = document.getElementById("mobile-navigation");
+    const list = document.getElementById("navigation-list");
+    const listItems = list?.querySelectorAll("li");
```

Comment:
Should throw an error if list not present in the document

Reply:
So: const listItems = list.querySelectorAll("li");' is correct?
VSC underline 'list' called: is propably null.

Reply:
It's not VSC, it's typescript ;)
```ts
if (listItems == null) {
  // throw an error
}
```

Reply:
Or you can use the non-null assertion
```ts
    const list = document.getElementById("navigation-list")!;
```

Reply:
Applied